proposal/nested-map-functions.md

```diff
@@ -0,0 +1,393 @@
+same output as our merge example, we can set each key individually:
+
+```scss
+@use 'sass:map';
+
+$nav: map.set($nav, 'color', 'hover', 'search', green);
+$nav: map.set($nav, 'color', 'hover', 'logo', orange);
+```
+
+And finally, a new `deep-merge()` function in the `sass:map` module allows
+merging two or more nested maps. This works much like the existing `merge()`
+function, but when both maps have a nested-map at the same key, those nested
+maps are also merged:
+
+```scss
+@use 'sass:map';
+
+$nav: (
+  'bg': 'gray',
+  'color': (
+    'hover': (
+      'search': yellow,
+      'home': red,
+      'filter': blue,
+    ),
+  ),
+);
+
+$update: (
+  'bg': white,
+  'color': (
+    'hover': (
+      'search': green,
+      'logo': orange,
+    )
+  )
+);
+
+$nav: map.deep-merge($nav, $update);
+
+// $nav: (
+//   'bg': white,
+//   'color': (
+//     'hover': (
+//       'search': green,
+//       'home': red,
+//       'filter': blue,
+//       'logo': orange,
+//     ),
+//   ),
+// );
+```
+
+## Functions
+
+All new and modified functions are part of the `sass:map` built-in module.
+
+### `get()`
+
+This proposal updates the signature and behavior of the existing `get()`
+function.
+
+> This also affects the global `map-get()` function.
+
+```
+get($map, $key, $keys...)
+```
+
+> Intuitively, `get($map, $key1, $key2, $key3)` is equivalent to
+> `get(get(get($map, $key1), $key2), $key3)` with the exception that if any
+> intermediate value isn't a map or doesn't have the given key the whole
+> function returns `null` rather than throwing an error.
+
+* If `$map` is not a map, throw an error.
+
+* Let `child` be `$map`.
+
+* Let `keys` be a list containing `$key` followed by the elements of `$keys`.
+
+* For each element `key` in `keys`:
+
+  * If `child` is not a map, return `null`.
+
+  * If `child` contains a key that's `==` to `key`, set `child` to the value
+    associated with that key. Otherwise, return `null`.
+
+* Return `child`.
+
+### `has-key()`
+
+This proposal updates the signature and behavior of the existing `get()`
+function.
+
+> This also affects the global `map-has-key()` function.
+
+```
+has-key($map, $key, $keys...)
+```
+
+> Intuitively, `has-key($map, $key1, $key2, $key3)` is equivalent to
+> `has-key(get(get($map, $key1), $key2), $key3)` with the exception that if any
+> intermediate value isn't a map or doesn't have the given key the whole
+> function returns `false` rather than throwing an error.
+
+* If `$map` is not a map, throw an error.
+
+* Let `child` be `$map`.
+
+* Let `keys` be a list containing `$key` followed by the elements of `$keys`.
+
+* For each element `key` in `keys`:
+
+  * If `child` is not a map, return `false`.
+
+  * If `child` contains a key that's `==` to `key`, set `child` to the value
+    associated with that key. Otherwise, return `false`.
+
+* Return `true`.
+
+### `set()`
+
+> Note: For consistency with other functions whose multi-key overloads were
```

Comment:
Since `map.get` and `map.has-key` no longer use overloads, the overload here isn't really for consistency anymore.

Would it make sense to just integrate the base case into the main `set` definition?

Reply:
It's consistent with `map.merge()`, which is the only other function that has a variable list of keys *before* a named final argument.

Reply:
Oh, right. That makes sense then